anta/runner.py

```diff
@@ -56,84 +71,126 @@ def log_cache_statistics(devices: list[AntaDevice]) -> None:
-    catalog: AntaCatalog,
-    devices: set[str] | None = None,
-    tests: set[str] | None = None,
-    tags: set[str] | None = None,
-    *,
-    established_only: bool = True,
-) -> None:
-    # pylint: disable=too-many-arguments
-    """Run ANTA.
-
-    Use this as an entrypoint to the test framework in your script.
-    ResultManager object gets updated with the test results.
+async def setup_inventory(inventory: AntaInventory, tags: set[str] | None, devices: set[str] | None, *, established_only: bool) -> AntaInventory | None:
+    """Set up the inventory for the ANTA run.
 
     Args:
     ----
-        manager: ResultManager object to populate with the test results.
         inventory: AntaInventory object that includes the device(s).
-        catalog: AntaCatalog object that includes the list of tests.
-        devices: devices on which to run tests. None means all devices.
-        tests: tests to run against devices. None means all tests.
         tags: Tags to filter devices from the inventory.
-        established_only: Include only established device(s).
-    """
-    limits = resource.getrlimit(resource.RLIMIT_NOFILE)
-    logger.debug("Initial limit numbers for open file descriptors for the current ANTA process: Soft Limit: %s | Hard Limit: %s", limits[0], limits[1])
-    nofile = __NOFILE__ if limits[1] > __NOFILE__ else limits[1]
-    logger.debug("Setting soft limit for open file descriptors for the current ANTA process to %s", nofile)
-    resource.setrlimit(resource.RLIMIT_NOFILE, (nofile, limits[1]))
-    limits = resource.getrlimit(resource.RLIMIT_NOFILE)
+        devices: Devices on which to run tests. None means all devices.
 
-    if not catalog.tests:
-        logger.info("The list of tests is empty, exiting")
-        return
+    Returns
+    -------
+        AntaInventory | None: The filtered inventory or None if there are no devices to run tests on.
+    """
     if len(inventory) == 0:
         logger.info("The inventory is empty, exiting")
-        return
+        return None
 
-    # Filter the inventory based on tags and devices parameters
-    selected_inventory = inventory.get_inventory(
-        tags=tags,
-        devices=devices,
-    )
+    # Filter the inventory based on the CLI provided tags and devices if any
+    selected_inventory = inventory.get_inventory(tags=tags, devices=devices) if tags or devices else inventory
+
+    # Connect to the devices
     await selected_inventory.connect_inventory()
 
     # Remove devices that are unreachable
     inventory = selected_inventory.get_inventory(established_only=established_only)
 
+    # If there are no devices in the inventory after filtering, exit
     if not inventory.devices:
         msg = f'No reachable device {f"matching the tags {tags} " if tags else ""}was found.{f" Selected devices: {devices} " if devices is not None else ""}'
         logger.warning(msg)
-        return
-    coros = []
+        return None
+
+    return selected_inventory
```

Comment:
```suggestion
    return inventory
```
Almost missed that one